seml/start.py

```diff
@@ -183,25 +214,31 @@ def start_slurm_job(collection, exp_array, unobserved=False, post_mortem=False,
         path = f"/tmp/{random_int}.sh"
     with open(path, "w") as f:
         f.write(script)
-    output = subprocess.check_output(f'sbatch {path}', shell=True)
-    slurm_array_job_id = int(output.split(b' ')[-1])
-    for task_id, chunk in enumerate(exp_array):
-        for exp in chunk:
-            if not unobserved:
-                collection.update_one(
-                        {'_id': exp['_id']},
-                        {'$set': {
-                            'status': States.PENDING[0],
-                            'slurm.array_id': slurm_array_job_id,
-                            'slurm.task_id': task_id,
-                            'slurm.sbatch_options': sbatch_options,
-                            'seml.output_file': f"{output_dir_path}/{name}_{slurm_array_job_id}_{task_id}.out"}})
-            logging.verbose(f"Started experiment with array job ID {slurm_array_job_id}, task ID {task_id}.")
+    if srun:
+        assert len(exp_array) == 1
+        assert len(exp_array[0]) == 1
+        cmd = (f"srun{srun_options_str} seml {' '.join(sys.argv[1:])} --local --sacred-id {exp_array[0][0]['_id']}")
+        subprocess.run(cmd, shell=True, check=True)
```

Comment:
Does it make sense to write the Slurm job ID into the database once `srun` started? Is it even possible?

Reply:
Done. See b440cfdc